pkg/tests/ossm/initContainer_test.go

```diff
@@ -15,42 +15,73 @@
 package ossm
 
 import (
-	_ "embed"
-	"strings"
+	"fmt"
 	"testing"
 
-	"github.com/maistra/maistra-test-tool/pkg/util"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
 const (
-	initContainerNS         = "bookinfo"
-	initContainerGoldString = "init worked"
+	initContainerNS          = "bookinfo"
+	initContainerGoldString  = "[init worked]"
+	initContainerPodSelector = "app=sleep-init"
 )
 
-var (
-	//go:embed yaml/deployment-sleep-init.yaml
-	testInitContainerYAML string
-)
+func TestInitContainer(t *testing.T) {
+	test.NewTest(t).Id("T33").Groups(test.Full).Run(func(t test.TestHelper) {
 
-func cleanupInitContainer() {
-	util.KubeDeleteContents(initContainerNS, testInitContainerYAML)
-}
+		hack.DisableLogrusForThisTest(t)
 
-func TestInitContainer(t *testing.T) {
-	test.NewTest(t).Id("T33").Groups(test.Full).NotRefactoredYet()
+		t.Cleanup(func() {
+			oc.DeleteFromString(t, initContainerNS, testInitContainerYAML)
+		})
 
-	defer cleanupInitContainer()
+		oc.ApplyString(t, initContainerNS, testInitContainerYAML)
 
-	if err := util.KubeApplyContents(initContainerNS, testInitContainerYAML); err != nil {
-		t.Fatalf("error creating the pod: %v", err)
-	}
-	if err := util.CheckPodRunning(initContainerNS, "app=sleep-init"); err != nil {
-		t.Fatalf("sleep-init pod is not running: %v", err)
-	}
+		oc.WaitPodReady(t, pod.MatchingSelector(initContainerPodSelector, initContainerNS))
 
-	logs := util.GetPodLogsForLabel(initContainerNS, "app=sleep-init", "init", false, false)
-	if !strings.Contains(logs, initContainerGoldString) {
-		t.Fatalf("expected init container log to contain the string %q, but got %q", initContainerGoldString, logs)
-	}
+		oc.Logs(t,
+			pod.MatchingSelector(initContainerPodSelector, initContainerNS),
+			"init",
+			assert.OutputContains(initContainerGoldString,
+				fmt.Sprintf("Log contains \"%s\".", initContainerGoldString),
+				fmt.Sprintf("Log does not contain \"%s\".", initContainerGoldString)))
```

Comment:
Here I'd ask myself what's more important. Not duplicating the string `[init worked]` or keeping the code short? I'd go with the latter here. By not worrying about duplication, this code becomes much simpler (and faster to read):
```
"Log contains '[init worked']"
```
instead of:
```
fmt.Sprintf("Log contains \"%s\".", initContainerGoldString),
```

Reply:
Also, according to the message of the commit this was implemented in the purpose of this test is: 

> This case tests if the init container is not removed when injecting the sidecar proxy.

So, a better success/failure message would probably be "init container is still present"/"init container did not run" or something like that.